airbyte-webapp/src/hooks/services/useConnectorAuth.tsx

```diff
@@ -105,18 +111,46 @@ export function useRunOauthFlow(
   done?: boolean;
   run: () => void;
 } {
-  const { values, setValues } = useFormikContext();
+  const {
+    values,
+    setValues,
+    validateForm,
+    setFieldTouched,
+  } = useFormikContext();
+  const oAuthFieldsToValidate = useMemo(() => {
+    const oAuthRequired =
+      connector?.authSpecification?.oauth2Specification?.oauthFlowInputFields;
+    return oAuthRequired ? oAuthRequired.map((value) => value.join(".")) : [];
+  }, [connector]);
+
+  const getoAuthInputParameters = (values: ServiceFormValues) => {
+    return pick(values.connectionConfiguration, oAuthFieldsToValidate);
+  };
   const { getConsentUrl, completeOauthRequest } = useConnectorAuth();
   const param = useRef<
     SourceGetConsentPayload | DestinationGetConsentPayload
   >();
 
   const [{ loading }, onStartOauth] = useAsyncFn(async () => {
-    const consentRequestInProgress = await getConsentUrl(connector);
-
-    param.current = consentRequestInProgress.payload;
-    openWindow(consentRequestInProgress.consentUrl);
-  }, [connector]);
+    const errors = (await validateForm(values)) as ServiceFormValues;
+    const oAuthErrors = Object.keys(
+      flatten(errors.connectionConfiguration)
+    ).filter((path) => oAuthFieldsToValidate.indexOf(path) != -1);
```

Comment:
:art: Suggestions for a bit better code style:

```suggestion
    ).filter((path) => oAuthFieldsToValidate.includes(path));
```

Reply:
Actually I am going to experiment with setTouched and validate: true. Then we won't need to check for includes. 

But thanks for your proposal!